swatch-contracts/ct/java/helpers/ContractsTestHelper.java

```diff
@@ -0,0 +1,134 @@
+import com.redhat.swatch.contract.test.model.RhEntitlementV1;
+import com.redhat.swatch.contract.test.model.SaasContractV1;
+import dto.ContractTestData;
+import io.restassured.response.Response;
+import java.util.List;
+import java.util.Objects;
+
+/** Helper class for contract related operations in component tests. */
+public final class ContractsTestHelper {
+
+  private static final String CREATE_CONTRACT_ENDPOINT = "/api/swatch-contracts/internal/contracts";
+  private static final String GET_CONTRACTS_ENDPOINT = "/api/swatch-contracts/internal/contracts";
+
+  private ContractsTestHelper() {}
+
+  public static ContractRequest buildContractRequest(ContractTestData contractData) {
+    Objects.requireNonNull(contractData, "contractData must not be null");
+
+    PartnerEntitlementV1 partnerEntitlement = buildPartnerEntitlement(contractData);
+
+    return new ContractRequest()
+        .partnerEntitlement(partnerEntitlement)
+        .subscriptionId(contractData.getSubscriptionId());
+  }
+
+  private static PartnerEntitlementV1 buildPartnerEntitlement(ContractTestData contractData) {
+    return new PartnerEntitlementV1()
+        .rhAccountId(contractData.getOrgId())
+        .sourcePartner(contractData.getSourcePartner())
+        .entitlementDates(buildEntitlementDates(contractData))
+        .rhEntitlements(List.of(buildRhEntitlement(contractData)))
+        .purchase(buildPurchase(contractData))
+        .partnerIdentities(buildPartnerIdentities(contractData));
+  }
+
+  private static DimensionV1 buildDimension(ContractTestData contractData) {
+    return new DimensionV1()
+        .name(contractData.getMetricName())
+        .value(contractData.getMetricValue());
+  }
+
+  private static PurchaseV1 buildPurchase(ContractTestData contractData) {
+    SaasContractV1 saasContract =
+        new SaasContractV1().dimensions(List.of(buildDimension(contractData)));
+
+    return new PurchaseV1()
+        .vendorProductCode(contractData.getProductCode())
+        .contracts(List.of(saasContract));
+  }
+
+  private static PartnerIdentityV1 buildPartnerIdentities(ContractTestData contractData) {
+    return new PartnerIdentityV1()
+        .awsCustomerId(contractData.getAwsCustomerId())
+        .sellerAccountId(contractData.getSellerAccountId())
+        .customerAwsAccountId(contractData.getAwsAccountId());
+  }
+
+  private static PartnerEntitlementV1EntitlementDates buildEntitlementDates(
+      ContractTestData contractData) {
+    return new PartnerEntitlementV1EntitlementDates()
+        .startDate(contractData.getStartDate())
+        .endDate(contractData.getEndDate());
+  }
+
+  private static RhEntitlementV1 buildRhEntitlement(ContractTestData contractData) {
+    return new RhEntitlementV1()
+        .subscriptionNumber(contractData.getSubscriptionNumber())
+        .sku(contractData.getSku());
+  }
+
+  public static Response createContract(SwatchService service, ContractTestData contractData) {
```

Comment:
This method should be in ContractsSwatchService.

Reply:
Done